pkgs/development/libraries/opencl-clang/default.nix

```diff
@@ -0,0 +1,110 @@
+{ stdenv
+, fetchFromGitHub
+, fetchpatch
+, cmake
+, git
+
+, llvmPackages_8
+, patch
+, runCommand
+, spirv-llvm-translator
+
+, buildWithPatches ? true
+}:
+
+let
+  llvmPkgs = llvmPackages_8 // {
+    inherit spirv-llvm-translator;
+  };
+
+  inherit (stdenv.lib) getVersion;
+
+  addPatches = component: pkg:
+    with builtins; with stdenv.lib;
+    let path = "${passthru.patchesOut}/${component}";
+    in pkg.overrideAttrs (super: {
+      postPatch = (if super ? postPatch then super.postPatch + "\n" else "") + ''
+        for p in ${path}/*
+        do
+          patch -p1 -i "$p"
+        done
+      '';
+    });
+
+  passthru = rec {
+
+    clang-unwrapped = addPatches "clang" llvmPkgs.clang-unwrapped;
+
+    clang = llvmPkgs.clang.override {
+      cc = clang-unwrapped;
+      cc-wrapper = runCommand "cc-wrapper" {
+        nativeBuildInputs = [ patch ];
+      } ''
+        patch -p1 -i ${./cc-wrapper.patch} -o $out ${../../../build-support/cc-wrapper/cc-wrapper.sh}
```

Comment:
I would rather include this in our cc-wrapper. Patching our own cc wrappers seems weird.

Reply:
After all it might be useful also outside of this project.

Reply:
Yup, that's what I intended, but I did it like this to make testing the PR possible without a mass rebuild. Once it's deemed otherwise ready I'll apply and remove the patch - or I can do it now if you prefer.

Reply:
ok, before we merge this. It should be applied and the branch changed to staging.